lib/client.js

```diff
@@ -467,7 +467,14 @@ function Request(method, url) {
       self.emit('response', res);
     }
 
-    self.callback(err, res);
+    if (res && res.status >= 200 && res.status < 300) {
```

Comment:
I know that redirect status codes are being treated before this line by superagent, but status codes >= 300 until < 400 are not http errors. :)